### PR TITLE
fix(contract): ETH conversion native hashes per chain

### PR DESCRIPTION
**Pull request by yomarion**
## Description of the changes

ChainlinkConversionPath v0.2.0
- Configure the native token hash in the constructor of Chainlink conversion path

ETHConversionProxy v0.2.0
- Whitelisted admins can edit the conversion path contract
- Whitelisted admins can edit the proxy contract

Live payment script:
- nonce 10: ChainlinkConversionPath v0.2.0
- nonce 11: ETHConversionProxy v0.2.0

### CODE REVIEW

File: packages/smart-contracts/scripts/3_deploy_chainlink_contract.ts [hunk 1]
```diff
@@ -36,7 +36,7 @@ export default async function deploy(args: any, hre: HardhatRuntimeEnvironment)
 
   const conversionPathInstance = await (
     await hre.ethers.getContractFactory('ChainlinkConversionPath', deployer)
-  ).deploy();
+  ).deploy(ETH_hash);
```
Comment (yomarion) [inline packages/smart-contracts/scripts/3_deploy_chainlink_contract.ts:39]: This is a local deployment only, in local we use the same hash as on mainnet.

File: packages/smart-contracts/src/contracts/EthConversionProxy.sol [hunk 2]
```diff
@@ -120,4 +121,26 @@ contract EthConversionProxy is ReentrancyGuard {
         amountToPay = (_requestAmount * rate) / decimals;
         amountToPayInFees = (_feeAmount * rate) / decimals;
     }
+
+    /**
+     * @notice Update the conversion path contract used to fetch conversions
+     * @param _chainlinkConversionPathAddress address of the conversion path contract
+     */
+    function updateConversionPathAddress(address _chainlinkConversionPathAddress)
```
Comment (yomarion) [inline packages/smart-contracts/src/contracts/EthConversionProxy.sol:129]: @vrolland  ideally if you could have a look at these changes 🙏

File: packages/smart-contracts/src/contracts/ChainlinkConversionPath.sol [hunk 3]
```diff
@@ -1,226 +1,231 @@
...
-      emit AggregatorUpdated(_inputs[i], _outputs[i], _aggregators[i]);
+    uint256 constant PRECISION = 1e18;
+    uint256 constant NATIVE_TOKEN_DECIMALS = 18;
+    uint256 constant FIAT_DECIMALS = 8;
+    address public nativeTokenHash;
+
+    /**
+     * @param _nativeTokenHash hash of the native token
+     */
+    constructor(address _nativeTokenHash) {
```
Comment (yomarion) [inline packages/smart-contracts/src/contracts/ChainlinkConversionPath.sol:32]: @vrolland  ideally if you could have a look at these changes 🙏

File: packages/smart-contracts/src/contracts/ChainlinkConversionPath.sol [hunk 4]
```diff
@@ -1,226 +1,231 @@
...
-  {
-    require(_inputs.length == _outputs.length, "arrays must have the same length");
-    require(_inputs.length == _aggregators.length, "arrays must have the same length");
-
-    // For every conversions of the path
-    for (uint i; i < _inputs.length; i++) {
-      allAggregators[_inputs[i]][_outputs[i]] = _aggregators[i];
-      emit AggregatorUpdated(_inputs[i], _outputs[i], _aggregators[i]);
+    uint256 constant PRECISION = 1e18;
+    uint256 constant NATIVE_TOKEN_DECIMALS = 18;
```
Comment (alexandre-abrioux) [inline packages/smart-contracts/src/contracts/ChainlinkConversionPath.sol:25]: Can this be different for another chain? I can't find the spec for this.

Comment (yomarion) [reply packages/smart-contracts/src/contracts/ChainlinkConversionPath.sol:25]: It's not completely impossible but unlikely. If it happens we can anticipate and adapt the contract before we deploy it (either with a hardcoded value or add a constructor parameter)

**Review by alexandre-abrioux — APPROVED** [review]

File: packages/smart-contracts/scripts/deploy-payments.ts [hunk 5]
```diff
@@ -215,27 +228,38 @@ export async function deployAllPaymentContracts(
...
-      nonceCondition: 7,
-    });
+    if (!args.simulate) {
+      throw new Error('Missing implementation new version of ERC20 Conversion Proxy');
+    }
 
-    // Batch 4
-    await runDeploymentBatch_4(chainlinkInstanceAddress, ethFeeProxyAddress);
+    // Batch 2
+    await runDeploymentBatch_2(chainlinkInstance, erc20FeeProxyAddress);
```
Comment (leoslr) [inline packages/smart-contracts/scripts/deploy-payments.ts:262]: Maybe I missed something, but it seems that when running :
```javascript
await runDeploymentBatch_2(chainlinkInstance, erc20FeeProxyAddress);
```
You should already have a nonce greater than 10 as it is the nonce condition to deploy the ChainlinkConversionPath contract. However there are several hardcoded nonce conditions lower than 10 in this function.

**Review by KolevDarko — APPROVED** [review]

File: packages/smart-contracts/scripts/deploy-payments.ts (hunk 5 above)
Comment (yomarion) [reply packages/smart-contracts/scripts/deploy-payments.ts:262]: That's why I added the error thrown just above, line 258, I'll fix that later but now I want to solve the incident quickly.

**Review by vrolland — COMMENTED** [review]
LGTM

**Pull request merged**